src/Compilers/CSharp/Portable/Symbols/Symbol.cs

```diff
@@ -349,6 +349,36 @@ public Location GetFirstLocation()
         public Location GetFirstLocationOrNone()
             => TryGetFirstLocation() ?? Location.None;
 
+        /// <summary>
+        /// Determines if there is a location (see <see cref="Locations"/>) for this symbol whose span is in <paramref
+        /// name="tree"/> and is contained within <paramref name="declarationSpan"/>.  Subclasses can override this to
+        /// be more efficient if desired (especially if avoiding allocations of the <see cref="Locations"/> array is
+        /// desired).
+        /// </summary>
+        public virtual bool HasLocationContainedWithin(SyntaxTree tree, TextSpan declarationSpan, out bool wasZeroWidthMatch)
+        {
+            foreach (var loc in this.Locations)
+            {
+                if (IsLocationContainedWithin(loc, tree, declarationSpan, out wasZeroWidthMatch))
+                    return true;
+            }
+
+            wasZeroWidthMatch = false;
+            return false;
+        }
+
+        protected static bool IsLocationContainedWithin(Location loc, SyntaxTree tree, TextSpan declarationSpan, out bool wasZeroWidthMatch)
```

Comment:
Why is this API on `Symbol` and not `SyntaxTree`? It seems purely about the latter so confused why it's here.

Reply:
primarily because it's just called from something symbolic and really seems to be about a symbolic concept (how we think about symbol containment and spans).  I could move to SyntaxTree if you'd prefer :)

Reply:
Deja vu. I swear I made the same comment about a method like this recently, because the relationship between `loc` and `tree` felt unclear to me.

Reply:
I am open to keeping this method here, if we feel the coupling to "here's how we decide if a symbol is *located here*" is too strong. Consider if there are more distinct names which can be used for the "where we are looking" versus the "possible thing we found" parameters.